### PR TITLE
emoticon manifests map to objects with file/name

### PR DESCRIPTION
**Pull request by miyu**
```python
import json
from pprint import pprint
from collections import OrderedDict
from odictliteral import odict

with open('index.json') as data_file:
    data = json.load(data_file, object_pairs_hook=OrderedDict)

emoticons = data["emoticons"];

for key in emoticons.keys():
    emoticons[key] = odict[
        'file': emoticons[key],
        'name': odict[
            'en': key
        ]
    ]

with open('index.json', 'w') as outfile:
    s = json.dumps(data, sort_keys = False, indent = 4, separators = (",", ": "))
    outfile.write(s)
    #outfile.write(s.replace('\n', '\r\n'))

```

### CODE REVIEW

**Comment by miyu** [comment]
Travis dies a tragic death when compiling LWIP. Tests passing on MBP fwiw.

File: animals/index.json [hunk 1]
```diff
@@ -5,17 +5,62 @@
         "Copyright 2016 Ranks.com Inc. <http://emojione.com>",
         "Copyright 2016 Beam Interactive, Inc <https://beam.pro>",
```
Comment (MattJustMatt) [inline animals/index.json]: This should be changed to Microsoft

Comment (MattJustMatt) [inline animals/index.json]: And updated to 2017 ;)

Comment (miyu) [inline animals/index.json]: Changing: `Copyright 2016 Beam Interactive, Inc <https://beam.pro>` => `Copyright 2017 Microsoft <https://microsoft.com>`

**Comment by miyu** [comment]
Btws, manifest output now looks like this:
```
  "animals": {
    "name": "Animals Emoticon Pack",
    "default": true,
    "authors": [ ... ],
    "emoticons": {
      ":fish": {
        "name": {
          "en": "fish"
        },
        "rect": {
          "x": 0,
          "y": 0,
          "width": 24,
          "height": 24
        }
      },
```

File: default/index.json [hunk 2]
```diff
@@ -3,134 +3,764 @@
...
+        "}:)": {
+            "file": "1F608",
+            "name": {
+                "en": "}:)"
+            }
+        },
+        "}:]": {
+            "file": "1F608",
+            "name": {
+                "en": "}:]"
```
Comment (connor4312) [inline default/index.json]: Need real English names for this ;)

File: extras/index.json [hunk 3]
```diff
@@ -3,21 +3,86 @@
...
+        ":facepalm": {
+            "file": "facepalm",
+            "name": {
+                "en": "facepalm"
+            }
+        },
+        ":rmf": {
+            "file": "rmf",
+            "name": {
+                "en": "rmf"
```
Comment (connor4312) [inline extras/index.json]: Probably won't screen read very well

File: memes/index.json [hunk 4]
```diff
@@ -3,17 +3,62 @@
...
-        ":cactus": "cactus",
-        ":mappa": "mappa",
-        ":suit": "suit",
-        ":wat": "wat",
-        ":yolo": "yolo",
-        ":whoappa": "whoappa"
+        ":127": {
+            "file": "127",
+            "name": {
+                "en": ":127"
```
Comment (connor4312) [inline memes/index.json]: Same

File: memes/index.json [hunk 5]
```diff
@@ -3,17 +3,62 @@
...
+            }
+        },
+        ":arrowknee": {
+            "file": "arrowknee",
+            "name": {
+                "en": "arrowknee"
```
Comment (connor4312) [inline memes/index.json]: And same

**Comment by miyu** [comment]
Changing spec by renaming "name" to "alt" for clarity.

File: default/index.json (hunk 2 above)
Comment (miyu) [inline default/index.json]: Using the emoji codepoints (e.g. "\u{1F636}" for 😶) instead. Screen readers support them: https://twitter.com/tenderlove/status/435480908687491072?lang=en

**Comment by miyu** [comment]
@MattJustMatt @connor4312

**Comment by connor4312** [comment]
Lgtm, assuming you confirmed that the narrator likes surrogate pairs?

**Comment by miyu** [comment]
Confirmed on chrome/macos with:
```
var msg = new SpeechSynthesisUtterance('\uD83D\uDC94');
window.speechSynthesis.speak(msg);
```

**Comment by connor4312** [comment]
I think the narrator may be different from that though, I would confirm with the guys on that email chain. Approving pending that 😄

**Comment by miyu** [comment]
Confirmed on edge/win10.

**Comment by miyu** [comment]
Worthwhile to note you're advised not to use unicode codepoints over surrogate pairs in json documents. Not sure on the specifics, but http://stackoverflow.com/questions/38463038/why-does-json-encode-utf-16-surrogate-pairs-instead-of-unicode-code-points-direc

**Comment by miyu** [comment]
also, manifest output is now
```
{
  "animals": {
    "name": "Animals Emoticon Pack",
    "default": true,
    "authors": [ ... ],
    "emoticons": {
      ":fish": {
          "x": 0,
          "y": 0,
          "width": 24,
          "height": 24,
          "alt": {
            "en": "fish"
          },
        }
      },
```

**Comment by connor4312** [comment]
> Worthwhile to note you're advised not to use unicode codepoints over surrogate pairs in json documents.

Yea, JSON parses are [surprisingly inconsistent](http://seriot.ch/parsing_json.php) and some can get tripped up by more edgy things like this.

**Pull request merged**